### ISSUE TITLE
[libsamplerate:x64-osx] build failure

### ISSUE THREAD

**Issue by monocasual** [microsoft/vcpkg#12915]
**Host Environment**
- OS: macOS 10.15.6
- Compiler: Xcode 11.6

**To Reproduce**
`./vcpkg install libsamplerate`

**Failure logs** 
Console output:
```
Computing installation plan...
The following packages will be built and installed:
    libsamplerate[core]:x64-osx
Detecting compiler hash for triplet x64-osx...
Starting package 1/1: libsamplerate:x64-osx
Building package libsamplerate[core]:x64-osx...
Could not locate cached archive: /Users/mcl/.cache/vcpkg/archives/85/855a21fe34881be71807ba46e96fad0b13c01640.zip
-- Using cached /Users/mcl/Software/vcpkg/downloads/libsamplerate-0.1.9.tar.gz
-- Cleaning sources at /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/mcl/Software/vcpkg/downloads/libsamplerate-0.1.9.tar.gz
-- Using source at /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean
-- Configuring x64-osx-dbg
-- Configuring x64-osx-rel
-- Building x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:139 (message):
    Command failed: /Users/mcl/Software/vcpkg/downloads/tools/cmake-3.17.2-osx/cmake-3.17.2-Darwin-x86_64/CMake.app/Contents/bin/cmake --build . --config Debug --target install -- -v
    Working Directory: /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/x64-osx-dbg
    See logs for more information:
      /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/install-x64-osx-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/libsamplerate/portfile.cmake:22 (vcpkg_install_cmake)
  scripts/ports.cmake:79 (include)


Error: Building package libsamplerate:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libsamplerate:x64-osx
  Vcpkg version: 2020.06.15-unknownhash

Additionally, attach any relevant sections from the log files above.
```

Additional `install-x64-osx-dbg-out.log`:
```
[1/6] /Users/mcl/Software/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc   -fPIC -g -isysroot /Users/mcl/Software/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -MD -MT CMakeFiles/libsamplerate-0.dir/src_zoh.o -MF CMakeFiles/libsamplerate-0.dir/src_zoh.o.d -o CMakeFiles/libsamplerate-0.dir/src_zoh.o   -c /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/src_zoh.c
FAILED: CMakeFiles/libsamplerate-0.dir/src_zoh.o 
/Users/mcl/Software/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc   -fPIC -g -isysroot /Users/mcl/Software/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -MD -MT CMakeFiles/libsamplerate-0.dir/src_zoh.o -MF CMakeFiles/libsamplerate-0.dir/src_zoh.o.d -o CMakeFiles/libsamplerate-0.dir/src_zoh.o   -c /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/src_zoh.c
In file included from /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/src_zoh.c:14:
/Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/float_cast.h:246:5: error: invalid instruction mnemonic 'fctiw'
                (       "fctiw  %1, %1\n\t"
                        ^
<inline asm>:1:2: note: instantiated into assembly here
        fctiw   %st, %st
        ^~~~~
In file included from /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/src_zoh.c:14:
/Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/float_cast.h:246:20: error: invalid instruction mnemonic 'stfd'
                (       "fctiw  %1, %1\n\t"
                                        ^
<inline asm>:2:2: note: instantiated into assembly here
        stfd    %st, -16(%rbp)
        ^~~~
2 errors generated.
[2/6] /Users/mcl/Software/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc   -fPIC -g -isysroot /Users/mcl/Software/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -MD -MT CMakeFiles/libsamplerate-0.dir/samplerate.o -MF CMakeFiles/libsamplerate-0.dir/samplerate.o.d -o CMakeFiles/libsamplerate-0.dir/samplerate.o   -c /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/samplerate.c
FAILED: CMakeFiles/libsamplerate-0.dir/samplerate.o 
/Users/mcl/Software/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc   -fPIC -g -isysroot /Users/mcl/Software/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -MD -MT CMakeFiles/libsamplerate-0.dir/samplerate.o -MF CMakeFiles/libsamplerate-0.dir/samplerate.o.d -o CMakeFiles/libsamplerate-0.dir/samplerate.o   -c /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/samplerate.c
In file included from /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/samplerate.c:16:
/Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/float_cast.h:246:5: error: invalid instruction mnemonic 'fctiw'
                (       "fctiw  %1, %1\n\t"
                        ^
<inline asm>:1:2: note: instantiated into assembly here
        fctiw   %st, %st
        ^~~~~
In file included from /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/samplerate.c:16:
/Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/float_cast.h:246:20: error: invalid instruction mnemonic 'stfd'
                (       "fctiw  %1, %1\n\t"
                                        ^
<inline asm>:2:2: note: instantiated into assembly here
        stfd    %st, -16(%rbp)
        ^~~~
2 errors generated.
[3/6] /Users/mcl/Software/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc   -fPIC -g -isysroot /Users/mcl/Software/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -MD -MT CMakeFiles/libsamplerate-0.dir/src_sinc.o -MF CMakeFiles/libsamplerate-0.dir/src_sinc.o.d -o CMakeFiles/libsamplerate-0.dir/src_sinc.o   -c /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/src_sinc.c
FAILED: CMakeFiles/libsamplerate-0.dir/src_sinc.o 
/Users/mcl/Software/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc   -fPIC -g -isysroot /Users/mcl/Software/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -MD -MT CMakeFiles/libsamplerate-0.dir/src_sinc.o -MF CMakeFiles/libsamplerate-0.dir/src_sinc.o.d -o CMakeFiles/libsamplerate-0.dir/src_sinc.o   -c /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/src_sinc.c
In file included from /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/src_sinc.c:14:
/Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/float_cast.h:246:5: error: invalid instruction mnemonic 'fctiw'
                (       "fctiw  %1, %1\n\t"
                        ^
<inline asm>:1:2: note: instantiated into assembly here
        fctiw   %st, %st
        ^~~~~
In file included from /Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/src_sinc.c:14:
/Users/mcl/Software/vcpkg/buildtrees/libsamplerate/src/mplerate-0-023220d426.clean/src/float_cast.h:246:20: error: invalid instruction mnemonic 'stfd'
                (       "fctiw  %1, %1\n\t"
                                        ^
<inline asm>:2:2: note: instantiated into assembly here
        stfd    %st, -16(%rbp)
        ^~~~
2 errors generated.
ninja: build stopped: subcommand failed.
```

**Additional context**
None.

**Comment by NancyLi1013** [comment]
Hi @monocasual 
Thanks for posting this issue.
Since `libsamplerate ` currently cannot build on osx. The results are as expected now.

Need to add support on osx in the future.

**Comment by RichLogan** [comment]
I had the same issue but I see vcpkg now tracks 0.2.1 which builds correctly so this can probably be closed.

**Comment by NancyLi1013** [comment]
Thanks for your kindly reminder @RichLogan.

Confirmed it has been solved in the latest version 0.2.1.
